### PR TITLE
fixes #210. Auth::user() return type is nullable

### LINKED ISSUE #210: Auth::user() can return null
- Larastan Version: v0.3.10
- `--level` used:

### Description:
`php artisan code:analyse` says
> If condition is always true.

However, the return type for `Auth::user()` should be nullable https://github.com/laravel/framework/blob/5.7/src/Illuminate/Support/Facades/Auth.php#L10

### Laravel code where the issue was found:
```
if (Auth::user()) {
            return redirect('/home');
}
return view('welcome');
```

### PR DESCRIPTION
**Pull request by mr-feek**
https://github.com/nunomaduro/larastan/issues/210

### CODE REVIEW

File: tests/Features/ReturnTypes/AuthExtension.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 class AuthExtension
 {
-    public function testUser(): User
+    public function testUser(): ?User
```
Comment (mr-feek) [inline tests/Features/ReturnTypes/AuthExtension.php]: This isn't a proper regression test, but I'm not familiar with the testing strategy used here and haven't had a chance to dig into it.

File: src/ReturnTypes/AuthExtension.php [hunk 2]
```diff
@@ -59,7 +59,7 @@ public function getTypeFromStaticMethodCall(
             ->get('config');
 
         if ($userModel = $config->get('auth.providers.users.model')) {
-            return new ObjectType($userModel);
+            return TypeCombinator::addNull(new ObjectType($userModel));
```
Comment (nunomaduro) [inline src/ReturnTypes/AuthExtension.php]: When the user is not logged, the `User::auth()` returns `null`?

Comment (mr-feek) [reply src/ReturnTypes/AuthExtension.php]: Correct, here is an example in the default `SessionGuard` that laravel provides: https://github.com/laravel/framework/blob/5.7/src/Illuminate/Auth/SessionGuard.php#L115-L117

**Comment by nunomaduro** [comment]
Perfect. Thanks!

**Pull request merged**